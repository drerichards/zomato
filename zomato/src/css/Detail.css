.Detail {
    flex: 1 1 45%;
    padding: 5% 1% 1%;
    height: 70vh;
}

.Detail .card {
    padding: 2%;
    overflow: hidden;
    max-height: 100%;
}

.Detail .hide {
    display: none;
}

.Detail #reviews hr {
    border: 0.25px ridge #fff;
    margin: 1% 0;
}

.Detail .tabs .tab a:hover,
.tabs .tab a.active {
    color: #F25F5C;
    font-weight: bold;
}

.Detail .card-bottom {
    background-color: #eabf53;
    max-height: 35em;
    overflow: auto;
}

.Detail .card .card-content p {
    font-size: 1.25rem;
    color: rgb(58, 30, 45);
}

.Detail .card .card-content p span {
    text-shadow: -1px 0px 1px rgb(192, 191, 191);
    color: #fff;
}

.Detail .tabs .indicator {
    height: 5px;
    background-color: #f4797e;
}

.Detail .reviewDiv .card .card-title {
    color: #000;
    text-align: right;
    font-weight: 700;
}

.Detail .reviewDiv .card {
    padding: 2% 2% 0;
}

.Detail .reviewDiv .reviewScroll {
    overflow: auto;
}

.Detail .reviewDiv .card .card-content {
    padding: 1% 2%;
}

.Detail .reviewDiv .card .card-content>p:nth-child(3) {
    text-align: right;
}

.Detail .reviewDiv .card .card-action {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 3% 0;
}

.Detail .reviewDiv .card .card-action p {
    font-size: 0.9em;
}