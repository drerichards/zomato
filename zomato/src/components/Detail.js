import React from 'react'
import PropTypes from 'prop-types'
import '../css/Detail.css'

const Detail = ({ name, address, city, votes, dollars, avg_cost_two, reservation, online_deliv, cuisines, rating_num, rating_text, url, reviews }) => {
    function dollarSign() {
        let dollarDiv = document.getElementById('dollar')
        dollarDiv.innerHTML = ''
        let elem = document.createElement('div')
        elem.setAttribute('style', 'display: inline-block;')
        dollarDiv.appendChild(elem)
        for (let i = 0; i < dollars; i++) {
            elem.innerHTML += '$'
        }
    }

    function renderReviews() {
        return reviews.map(function (review, i) {
            return <div key={i} className='card'>
                <div className="card-content white-text">
                    <span className="card-title">''{review.rating_title}''</span>
                    <p>{review.review_text}</p>
                    <p>- {review.user}</p>
                    <div className="card-action">
                        <p>Rating: {review.rating}</p>
                        <p>{review.review_time}</p>
                    </div>
                </div>
            </div>
        })
    }

    return (
        <div className='Detail'>
            <div className={name ? 'card' : 'card hide'}>
                <div className="card-content">
                    <h4>{name}</h4>
                    <hr />
                </div>
                <div className="card-tabs">
                    <ul className="tabs tabs-fixed-width">
                        <li className="tab"><a href="#location" className='active'>Location</a></li>
                        <li className="tab"><a href="#info">Info</a></li>
                        <li className="tab"><a href="#reviews">Reviews</a></li>
                    </ul>
                </div>
                <div className="card-content card-bottom">
                    <div id="location">
                        <p>Address: <span>{address}</span></p>
                        <p>City: <span>{city}</span></p>
                    </div>
                    <div id="info">
                        <p>Cuisines: <span>{cuisines}</span></p>
                        <p>Price Range: <span id='dollar'>{dollars ? dollarSign() : ''}</span></p>
                        <p>Average cost for two diners: <span>${avg_cost_two}</span></p>
                        <p>Table Reservation: <span>{reservation === 0 ? 'No' : 'Yes'}</span></p>
                        <p>Online Delivery: <span>{online_deliv === 0 ? 'No' : 'Yes'}</span></p>
                        <p>Visit Website: <a href={url} target="_blank">{name}</a> </p>
                    </div>
                    <div id="reviews">
                        <p>Rating: <span>{rating_text}</span></p>
                        <p>Score: <span>{rating_num} out of 5</span></p>
                        <p>Votes: <span>{votes}</span></p>
                        <hr />
                        <section className="reviewDiv">
                            <h5>{reviews ? reviews.length : 'No'} Reviews:</h5>
                            <div className="reviewScroll">
                                {reviews ? renderReviews() : ''}
                            </div>
                        </section>
                    </div>
                </div>
            </div>
        </div>
    )
}

Detail.propTypes = {
    name: PropTypes.string,
    address: PropTypes.string,
    city: PropTypes.string,
    latitude: PropTypes.string,
    longitude: PropTypes.string,
    cuisines: PropTypes.string,
    rating_num: PropTypes.string,
    rating_text: PropTypes.string,
    votes: PropTypes.string,
    dollars: PropTypes.number,
    avg_cost_two: PropTypes.number,
    reservation: PropTypes.number,
    online_deliv: PropTypes.number,
    url: PropTypes.string,
    reviews: PropTypes.array
}
export default Detail
