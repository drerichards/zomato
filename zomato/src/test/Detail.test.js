import React from 'react'
import renderer from 'react-test-renderer'
import Detail from '../components/Detail'

test('Detail snapshot test', () => {
    const component = renderer.create(<Detail />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})