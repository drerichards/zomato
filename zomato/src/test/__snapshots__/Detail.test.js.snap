// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail snapshot test 1`] = `
<div
  className="Detail"
>
  <div
    className="card hide"
  >
    <div
      className="card-content"
    >
      <h4 />
      <hr />
    </div>
    <div
      className="card-tabs"
    >
      <ul
        className="tabs tabs-fixed-width"
      >
        <li
          className="tab"
        >
          <a
            className="active"
            href="#location"
          >
            Location
          </a>
        </li>
        <li
          className="tab"
        >
          <a
            href="#info"
          >
            Info
          </a>
        </li>
        <li
          className="tab"
        >
          <a
            href="#reviews"
          >
            Reviews
          </a>
        </li>
      </ul>
    </div>
    <div
      className="card-content card-bottom"
    >
      <div
        id="location"
      >
        <p>
          Address: 
          <span />
        </p>
        <p>
          City: 
          <span />
        </p>
      </div>
      <div
        id="info"
      >
        <p>
          Cuisines: 
          <span />
        </p>
        <p>
          Price Range: 
          <span
            id="dollar"
          >
            
          </span>
        </p>
        <p>
          Average cost for two diners: 
          <span>
            $
          </span>
        </p>
        <p>
          Table Reservation: 
          <span>
            Yes
          </span>
        </p>
        <p>
          Online Delivery: 
          <span>
            Yes
          </span>
        </p>
        <p>
          Visit Website: 
          <a
            href={undefined}
            target="_blank"
          />
           
        </p>
      </div>
      <div
        id="reviews"
      >
        <p>
          Rating: 
          <span />
        </p>
        <p>
          Score: 
          <span>
             out of 5
          </span>
        </p>
        <p>
          Votes: 
          <span />
        </p>
        <hr />
        <section
          className="reviewDiv"
        >
          <h5>
            No
             Reviews:
          </h5>
          <div
            className="reviewScroll"
          >
            
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
